crates/relayer-runtime/src/tokio/context.rs

```diff
@@ -11,39 +9,26 @@ use tokio::sync::{mpsc, oneshot};
 use tokio::time::sleep;
 use tracing;
 
-use ibc_relayer_framework::base::core::traits::sync::Async;
 use ibc_relayer_framework::base::one_for_all::traits::chain::OfaBaseChain;
 use ibc_relayer_framework::base::one_for_all::traits::runtime::{LogLevel, OfaRuntime};
 use ibc_relayer_framework::full::one_for_all::traits::batch::OfaBatch;
 
 use super::error::Error as TokioError;
 
-pub struct TokioRuntimeContext<Error> {
+#[derive(Clone)]
+pub struct TokioRuntimeContext {
     pub runtime: Arc<Runtime>,
-    pub phantom: PhantomData<Error>,
 }
 
-impl<Error> TokioRuntimeContext<Error> {
+impl TokioRuntimeContext {
     pub fn new(runtime: Arc<Runtime>) -> Self {
-        Self {
-            runtime,
-            phantom: PhantomData,
-        }
-    }
-}
-
-impl<Error> Clone for TokioRuntimeContext<Error> {
-    fn clone(&self) -> Self {
-        Self::new(self.runtime.clone())
+        Self { runtime }
     }
 }
 
 #[async_trait]
-impl<Error> OfaRuntime for TokioRuntimeContext<Error>
-where
-    Error: From<TokioError> + Debug + Async,
```

Comment:
Since the constraint `From<TokioError>` is removed, should the implementation, https://github.com/informalsystems/hermes/blob/soares/explicit-error-injection/crates/relayer-cosmos/src/base/error.rs#L59, be kept or not ?

Reply:
Yup, it can be removed if there is no compile error. Thanks for pointing out!